airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/cursor.py

```diff
@@ -58,6 +60,23 @@ def extract_value(self, record: Record) -> CursorValueType:
         return cursor_value  # type: ignore  # we assume that the value the path points at is a comparable
 
 
+class InterpolatedCursorField(CursorField):
```

Comment:
I've added this new `InterpolatedCursorField` so that we can perform interpolation evaluation at processing time when we observe records, instead of at parse time when we build the `ConcurrentCursor`.

I feel like this is a nice to have from the mental model of interpolation only being used at runtime, but the unfortunate side effect is that we leak interpolation as a concept into the Concurrent CDK. I've included the code to spark discussion, but my vote is that we remove this and just perform the interpolation ahead of time when we instantiate the `ConcurrentCursor`